### ISSUE TITLE
Only one task following an aggregation is executed

### ISSUE THREAD

**Issue by schuhschuh** [openmole/openmole#63]
Given the following test script, I would expect the computation and output of both the mean value as well as the standard deviation (sigma) of the values produced by main (actually the exploration in this case). However, only one of the values is computed and displayed!

```scala
val v = Val[Double]
val m = Val[Double]
val s = Val[Double]

def exploration = ExplorationTask(v in (UniformDistribution[Double]() take 10))

def main =
  EmptyTask() set (
    name    := "main",
    inputs  += v,
    outputs += v
  )

def mean =
  ScalaTask("val m = v.sum / v.size") set (
    name    := "mean",
    inputs  += v.toArray,
    outputs += m
  )

def sigma =
  ScalaTask(
    """
     | val m  = v.sum / v.size
     | val m2 = v.map(math.pow(_, 2)).sum / v.size
     | val s  = math.sqrt(m2 - m * m)
    """.stripMargin
  ) set (
    name    := "sigma",
    inputs  += v.toArray,
    outputs += s
  )

val mole = exploration -< main >- (mean hook DisplayHook("Mean = ${m}"), sigma hook DisplayHook("Sigma = ${s}"))
val ex   = mole.start
ex.waitUntilEnded
```

Output:
```
Loading ParallelTasksAfterAggregationTest.scala...
v: org.openmole.core.workflow.data.Prototype[Double] = v: Double
m: org.openmole.core.workflow.data.Prototype[Double] = m: Double
s: org.openmole.core.workflow.data.Prototype[Double] = s: Double
warning: there were three feature warnings; re-run with -feature for details
exploration: org.openmole.core.workflow.builder.TaskBuilder{def toTask: org.openmole.core.workflow.task.ExplorationTask with this.Built}
warning: there was one feature warning; re-run with -feature for details
main: org.openmole.core.workflow.builder.TaskBuilder{def toTask: org.openmole.core.workflow.task.EmptyTask with this.Built}
warning: there was one feature warning; re-run with -feature for details
mean: org.openmole.plugin.task.scala.ScalaTaskBuilder{def toTask: org.openmole.plugin.task.scala.ScalaTask with this.Built}
warning: there was one feature warning; re-run with -feature for details
sigma: org.openmole.plugin.task.scala.ScalaTaskBuilder{def toTask: org.openmole.plugin.task.scala.ScalaTask with this.Built}
mole: org.openmole.core.workflow.puzzle.Puzzle = Puzzle(Slot hash 1885447655 of explorationTask0,List(mean, sigma),List(AggregationTransition from main to Slot hash 2037727008 of mean, ExplorationTransition from explorationTask0 to Slot hash 1355326338 of main, AggregationTransition from main to Slot hash 78530047 of sigma, ExplorationTransition from explorationTask0 to Slot hash 1355326338 of main),List(),List(),List((mean,org.openmole.plugin.hook.display.DisplayHook$$anon$2$$anon$1@3cced001), (sigma,org.openmole.plugin.hook.display.DisplayHook$$anon$2$$anon$1@6c927487)),Map(),Map())
ex: org.openmole.core.workflow.mole.MoleExecution = org.openmole.core.workflow.mole.MoleExecution@22a7ff5d
Sigma = 0.29359506066346125
res0: org.openmole.core.workflow.mole.MoleExecution = org.openmole.core.workflow.mole.MoleExecution@22a7ff5d
```

**Comment by schuhschuh** [comment]
Note that the issue also persists when a capsule is used with separate puzzles merged using "+", i.e.,
```scala
val mainCap = Capsule(main)
def sub1 = exploration -< mainCap
def sub2 = mainCap >- (mean  hook DisplayHook("Mean  = ${m}"))
def sub3 = mainCap >- (sigma hook DisplayHook("Sigma = ${s}"))
val mole = sub1 + sub2 + sub3
```

**Comment by schuhschuh** [comment]
@romainreuillon  fixed it in 8c30775f29bb183096c7590f9331f38db1c73198.
